src/concurrency/sync.rs

```diff
@@ -66,6 +66,30 @@ pub(super) use declare_id;
 
 declare_id!(MutexId);
 
+/// The mutex kind.
+#[derive(Debug, Clone, Copy)]
+pub struct MutexKind(i32);
+
+impl MutexKind {
+    pub fn new(k: i32) -> Self {
+        Self(k)
+    }
+
+    pub fn to_i32(&self) -> i32 {
+        self.0
+    }
+}
+
+#[derive(Debug)]
+/// Additional data that may be used by shim implementations.
+pub struct AdditionalMutexData {
+    /// The mutex kind, used by some mutex implementations like pthreads mutexes.
+    pub kind: MutexKind,
+
+    /// The address of the mutex.
+    pub address: Pointer,
```

Comment:
Since we really only care about the address, this should be a `u64`. Currently you are also storing provenance but we don't actually care about that.